src/main/java/org/javarosa/xform/parse/FormInstanceParser.java

```diff
@@ -392,25 +392,23 @@ private void applyInstanceProperties(FormInstance instance) {
             EvaluationContext ec = new EvaluationContext(instance);
             final List<TreeReference> nodes = ec.expandReference(ref, true);
 
-            if (nodes.size() > 0) {
```

Comment:
It's not immediately clear why this is redundant. Is it just that if there was no node then there couldn't be a bind?

Reply:
Yes.

I'll revisit the commit in case I can leave more clues about this.